### PR TITLE
Implement proxy mechanism and propagate to the map viewer #105

### PR DESCRIPTION
**Pull request by napakalas**

### CODE REVIEW

**Comment by dbrnz** [comment]
Hmm, this is passing proxies to the viewer as part of a feature's properties -- I was more thinking of a proxies JSON file that gets embedded as part of the metadata in the `index.mbtiles` database for a map-server to provide to the viewer...

**Comment by napakalas** [comment]
So, all I have to do is save it using a code like:
`tile_db.add_metadata(proxies=json.dumps(proxy_features))`

where `proxy_features` is extracted from `proxy_features.json`, is that right?

**Comment by napakalas** [comment]
Do I need to keep the original proxy_features format in the metadata or convert it to format such as:
```
{
    "UBERON:0000044":["UBERON:0003943"]
}
```
where the key is a feature and the value is a list of poxy.

**Comment by dbrnz** [comment]
> So, all I have to do is save it using a code like:
tile_db.add_metadata(proxies=json.dumps(proxy_features))

Correct (and only if there are `proxy_features` for the map.

> Do I need to keep the original proxy_features format in the metadata or convert it to format such as:

Debatable, although my preference is to use appropriate and well defined names for keys as then the data becomes self-documenting and can be described by a schema or type declaration, etc. We only need `feature` and `proxies`; there's no need to pass `name` or other fields.

**Comment by napakalas** [comment]
The latest three commits:
- cancel proxies as parts of the feature's properties
- format proxies such as as
```
[
     {
           "feature":"UBERON:0000044",
           "proxies":["UBERON:0003943"]
     }
]
```
- save it to `index.mbtiles`

File: mapmaker/properties/__init__.py [hunk 1]
```diff
@@ -111,6 +111,15 @@ def __init__(self, flatmap: "FlatMap", manifest: "Manifest"):
         self.__networks = { network.get('id'): Network(flatmap, network, self)
                                 for network in properties_dict.get('networks', []) }
 
+        # Proxy features defined in JSON
+        self.__proxies = []
+        if manifest.proxy_features is not None:
+            for proxy in FilePath(manifest.proxy_features).get_json():
+                self.__proxies += [{
```
Comment (dbrnz) [inline mapmaker/properties/__init__.py:118]: How about:
```
# Proxy features defined in JSON
self.__proxies = ([dict(feature=proxy['feature'], proxies=proxy['proxies']) 
                    for proxy in FilePath(manifest.proxy_features).get_json()] 
                if manifest.proxy_features is not None 
                else [])
```

**Comment by dbrnz** [comment]
BTW are you aware these no need for a new PR when making changes -- simply push to your branch which was used to create the original PR and GitHub will take care of things.

File: mapmaker/properties/__init__.py (hunk 1 above)
Comment (napakalas) [reply mapmaker/properties/__init__.py]: Yes, I like it. It is more concise. I've pushed a commit.

**Comment by napakalas** [comment]
> BTW are you aware these no need for a new PR when making changes -- simply push to your branch which was used to create the original PR and GitHub will take care of things.

Yes I am aware about this. I have pushed a new commit for `self.__proxies` assignment.

**Pull request merged**